README.md

```diff
@@ -110,86 +110,122 @@ bash test/test_cleanup.sh
 
-The third and final workflow, **_QUANTIFY_** quantifies miRNA expression by
-intersecting the alignments from the **_MAP_** workflow with the annotations
-generated in the **_PREPARE_** workflow. Intersections are computed with
+The third and final subworkflow, **_QUANTIFY_**, quantifies miRNA expression by
+intersecting the alignments from the **_MAP_** subworkflow with the annotations
+generated in the **_PREPARE_** subworkflow. Intersections are computed with
 [`bedtools`][bedtools] for one or multiple of mature, primary transcripts and
 isomiRs. Reads consistent with each miRNA are counted and tabulated.
 
-The scheme below is a visual representation of an example run of the
-**_QUANTIFY_** workflow:
 
-> ![rule-graph-quantify][rule-graph-quantify]
+The schema below is a visual representation of
+an example run of the whole workflow:
+
+> ![rule-graph][rule-graph]
 
-### Running the workflows
 
-Assuming that you are currently inside the repository's root directory, change
-to the run root directory:
+### Running the workflow
+
+Assuming that you are currently inside the repository's root directory, 
+create a directory from which you will run your workflow and
+name it whatever you want e.g., `MY_ANALYSIS` and 
+head to it.
 
 ```bash
-cd RUNS
+mkdir MY_ANALYSIS
+cd MY_ANALYSIS
 ```
+Now within the `MY_ANALYSIS/` directory, 
+create a new directory to store you library; 
+name it as you like, e.g. `my_lib` and move there.    
 
-Now make a clean copy of the `JOB` directory and name it whatever you want,
-e.g., `MY_ANALYSIS`:
+```bash
+mkdir my_lib
+cd my_lib
+```
+Place your library file(s) here. 
+In addition, create a sample table. 
+Fill it with the correct entries. 
+You can look at the `test/test_files/sample_table.csv` 
+to get an idea of what this file must look like.
 
 ```bash
-cp -r JOB MY_ANALYSIS
+touch sample_table.csv
 ```
 
-Now traverse to the new directory. You will see that there are three
-subdirectories, one for each workflow, change into the one you would like to
-run (probably `prepare`).
+Now traverse back to the previous directory.
+Here, a couple of things must be done:  
+
+First of all, copy the `config_template.yaml` to this directory.
 
 ```bash
-cd MY_ANALYSIS
-cd {prepare,map,quantify}
+cp ..config/config_template.yaml ./config.yaml
+```
+Then, using your editor of choice, adjust the parameters of the `config.yaml`.
+The file explains what each of the parameters means and how you can meaningfully
+fill them in.  
+
+
+Accordingly to how you want
+to run the workflow you can either 
+copy the script to run it **locally** with
+
+```bash
+cp ../test/test_workflow_local.sh ./run_workflow_local.sh
+```
+or copy the script to run the workflow 
+on a **cluster via Slurm** 
+along with the `cluster.json` file with
+
+```bash
+cp ../test/test_workflow_slurm.sh ./run_workflow_slurm.sh
+cp ../test/cluster.json ./cluster.json
+```
+
+In both cases, the final 9 lines must be removed; this can be done with:
```

Comment:
Remove the last 9 lines? 😆 

Instead of telling the users in detail that they should copy and then modify and trim the files by 9 lines (will you remember to update this part of the docs if you just add another two lines to the test script in 3 months?), just tell them what inputs are expected, where examples can be found, and how the Snakemake commands look like to:

- Start a dry run that creates a rule graph (can replace the section above)
- Start an actual run locally
- Start a run on an HPC with Slurm as example
- Create a report

And use different subsections for these.

Reply:
Several things to finally get the sync documentation.

To start a local run, the user does not need to have a script, just to run the `snakemake` command, should I recommend to the command line on a script to be able to use it in different runs?  

Should the `test_cleanup.sh` should be recommended to be on the working directory in case some errors arise and must remove stuff?  

The creation of the report should be recommended to be added at the end of the run script so it is done altogether? This is, having a section on the `README.md` giving the command line to obtain a report but add the recommendation of having it altogether and provide an example of how the hole script should look like?

(still working on the cluster run, internet today is horribly slow)

Reply:
You don't need to explain to people how Bash/shell works. If they want to put it in a script, they can do that (or find out elsewhere how, if they don't know already). Providing convenience scripts can be useful, but in this case, not all that much, because you restrict people's ability to change params easily. That's what a CLI is for. And we don't wanna start and build a CLI for _mirflowz_, it's not worth the trouble.

So just stick to the bare commands. The testing script is for running tests - that's more for _us_. You can include a reference on how to run that at the end of the installation section as an optional step to check whether all seems to have been installed/deployed fine. But I don't think it should be part of the _usage_ instructions.